Index: common/unicode/pmac.h
===================================================================
--- common/unicode/pmac.h	(revision 69841)
+++ common/unicode/pmac.h	(working copy)
@@ -49,14 +49,20 @@
 #define U_DARWIN
 #endif
 
+#include <AvailabilityMacros.h>
+
 /**
  * \def U_HAVE_DIRENT_H
  * Define whether dirent.h is available 
  * @internal
  */
 #ifndef U_HAVE_DIRENT_H
+#if MAC_OS_X_VERSION_MIN_REQUIRED < MAC_OS_X_VERSION_10_5
+#define U_HAVE_DIRENT_H 0
+#else
 #define U_HAVE_DIRENT_H 1
 #endif
+#endif
 
 /** Define whether inttypes.h is available */
 #ifndef U_HAVE_INTTYPES_H
@@ -282,8 +288,10 @@
 #if 1
 #define U_TZSET         tzset
 #endif
-#if 1
-#define U_TIMEZONE      timezone
+#if MAC_OS_X_VERSION_MIN_REQUIRED < MAC_OS_X_VERSION_10_5
+#define U_TIMEZONE 0
+#else
+#define U_TIMEZONE timezone
 #endif
 #if 1
 #define U_TZNAME        tzname
